src/json_encoding.ml

```diff
@@ -498,20 +497,22 @@ let mu name self = Mu (name, self)
 let null = Null
 let int =
   Int { name = "int" ;
-        of_float = int_of_float ;
-        to_float = float_of_int ;
+        of_string = int_of_string ;
+        to_string = string_of_int ;
+        (* Why not min_int and max_int? *)
         lower_bound = -(1 lsl 30) ;
         upper_bound = (1 lsl 30) - 1 }
 let ranged_int ~minimum ~maximum name =
   Int { name ;
-        of_float = int_of_float ;
-        to_float = float_of_int ;
+        of_string = int_of_string ;
+        to_string = string_of_int ;
         lower_bound = minimum ;
         upper_bound = maximum }
 
 let ranged_float ~minimum ~maximum name =
-  Float (Some { minimum ;
-                maximum })
+  let minimum = string_of_float minimum in
+  let maximum = string_of_float maximum in
+  Float (Some { minimum ; maximum })
```

Comment:
I think that representing integers as strings is the right thing to do, but I am less convinced by the floating-point part of the change:

- `string_of_float` will lose information for some numbers, for example `string_of_float (1. +. 1e-12)` is just `"1."` on my machine, while the two numbers are distinct. 
- I have the impression that keeping the type as `float` as long as possible in the pipeline is beneficial/informative?

For lossless transport of floating-point numbers, the following representations would work:
- `Int64.bits_of_float` gives access to the raw bit patterns of the floating-point number --
  and can then be represented without loss as a string. (One downside is that it is unreadable for humans.)
- the hexadecimal representation of floats (see https://github.com/ocaml/ocaml/pull/268) should also be lossless, so you could use `Printf.sprintf "%x"` to serialize floats (then `float_of_string` should still work). For example, `1. +. 1e-12` gives `"0x1.0000000001198p+0"`

It might be possible to use a two-times approach, first using just `string_of_float`, checking for loss of information, and using `%x` for the corner cases only, if you want to favor readability of the output over serialization performance.

Reply:
It might even be better to leave all floats as floats and only use string for ints. I think this specific change is because the type for bounds is used for both ints and floats somewhere. Or maybe I went a bit overboard.

I'll try to do that.

Reply:
The goal of ocplib-json-typed is to provide JS friendly APIs.
Using strings for all integers is not.

I believe that we should only accept int ranges that fit into an int31 and keep using numbers (adding a dynamic test for the ranged_int on 64-bit platforms to restrict it to int31).
We could add an int64 combinator that uses strings for when we need larger integers.

That way we use the appropriate JS types, and remain consistent accross platforms.